### PR TITLE
Refactor SootUp module structure

### PR DESCRIPTION
**Pull request by stschott**
- Separate sootup.analysis into sootup.analysis.interprocedural and sootup.analysis.intraprocedural
- Separate body interceptors from sootup.java.core into new module sootup.interceptors
- Move Dex interceptors into sootup.apk.parser
- Move ReachingDefs into sootup.analysis.intraprocedural
- Add new modules to report and DependencyManagement section in root pom.xml 
- add new modules to gh actions workflow ;)

### CODE REVIEW

File: sootup.analysis.intraprocedural/pom.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.soot-oss</groupId>
+            <artifactId>sootup.java.sourcecode</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
```
Comment (JonasKlauke) [inline sootup.analysis.intraprocedural/pom.xml:32]: here it is missing the test scope

File: sootup.analysis.intraprocedural/pom.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        </dependency>
+    </dependencies>
+    <repositories>
+        <repository>
+            <id>sonatype-snapshots</id>
```
Comment (JonasKlauke) [inline sootup.analysis.intraprocedural/pom.xml]: why do we need the snapshot repo?

File: sootup.apk.parser/src/main/java/interceptors/AbstractNullTransformer.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package sootup.java.core.interceptors.Dex;
+package interceptors;
```
Comment (JonasKlauke) [inline sootup.apk.parser/src/main/java/interceptors/AbstractNullTransformer.java]: the package should be sootup.apk.parser.interceptors.

**Review by JonasKlauke — CHANGES_REQUESTED** [review]

**Comment by JonasKlauke** [comment]
I think we should also rename sootup.apk.parser and sootup.jimple.parser and drop the parser part

**Comment by swissiety** [comment]
> I think we should also rename sootup.apk.parser and sootup.jimple.parser and drop the parser part

Then we should split the concrete Jimple Implementations from core and only keep interfaces there.

**Review by JonasKlauke — APPROVED** [review]

**Pull request merged**